Add "etl-pull-queue" and sample insert queue command.

stephen-soltesz opened the pull request:
This change adds a pull queue configuration to the queue.yaml file.

As well, this change includes a sample task queue client that inserts a given payload into the pull queue.

gfr10598 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>  After comments resolved.

---

Review status: 0 of 2 files reviewed at latest revision, 6 unresolved discussions.

---

*[appengine/queue.yaml, line 12 at r1](https://reviewable.io:443/reviews/m-lab/etl/7#-KhOy2exwqRVf3Ok53aU:-KhOy2exwqRVf3Ok53aV:b-3qqes8) ([raw file](https://github.com/m-lab/etl/blob/0365c1e087f528304177ac212d895b6e0d6acf96/appengine/queue.yaml#L12)):*
> ```YAML
>   mode: pull
>   acl:
>   - user_email: 581276032543-compute@developer.gserviceaccount.com
> ```

Comment what VM this corresponds to.

---

*[cmd/task_client/task_client.go, line 23 at r1](https://reviewable.io:443/reviews/m-lab/etl/7#-KhOxWNqfU2xSFaClfkj:-KhOxWNqfU2xSFaClfkk:bnhfb22) ([raw file](https://github.com/m-lab/etl/blob/0365c1e087f528304177ac212d895b6e0d6acf96/cmd/task_client/task_client.go#L23)):*
> ```Go
> // Create taskqueue.Service for communicating with the taskqueue services.
> func CreateTaskQueueService() *taskqueue.Service {
> ```

Blank line?

---

*[cmd/task_client/task_client.go, line 24 at r1](https://reviewable.io:443/reviews/m-lab/etl/7#-KhOxYIloHTESTo4GlNB:-KhOxYIloHTESTo4GlNC:b-5hxsqy) ([raw file](https://github.com/m-lab/etl/blob/0365c1e087f528304177ac212d895b6e0d6acf96/cmd/task_client/task_client.go#L24)):*
> ```Go
> func CreateTaskQueueService() *taskqueue.Service {
> 
> 	// NOTE: DefaultClient uses "Application Default Credentials", which derive
> ```

This note isn't super clear.  Ok to leave as is, but if you can make it clearer that would be good.

---

*[cmd/task_client/task_client.go, line 44 at r1](https://reviewable.io:443/reviews/m-lab/etl/7#-KhOxFTdBjH-ea9I_-WR:-KhOxFTdBjH-ea9I_-WS:b-m9fy90) ([raw file](https://github.com/m-lab/etl/blob/0365c1e087f528304177ac212d895b6e0d6acf96/cmd/task_client/task_client.go#L44)):*
> ```Go
> func NewTask(queueName, payload string) *taskqueue.Task {
> 	t := &taskqueue.Task{
> 		// Set first available least time to now.
> ```

s/least/lease/

---

*[cmd/task_client/task_client.go, line 54 at r1](https://reviewable.io:443/reviews/m-lab/etl/7#-KhOxnfsssJQR913BNF1:-KhOxnfsssJQR913BNF2:b-7k4kmx) ([raw file](https://github.com/m-lab/etl/blob/0365c1e087f528304177ac212d895b6e0d6acf96/cmd/task_client/task_client.go#L54)):*
> ```Go
> func main() {
> 	api := CreateTaskQueueService()
> 	pretty.Print(api)
> ```

Nice - will have to remember this.

---

*[cmd/task_client/task_client.go, line 58 at r1](https://reviewable.io:443/reviews/m-lab/etl/7#-KhOxHCxdR9uRsCxY3BT:-KhOxHCxdR9uRsCxY3BU:b40p0zp) ([raw file](https://github.com/m-lab/etl/blob/0365c1e087f528304177ac212d895b6e0d6acf96/cmd/task_client/task_client.go#L58)):*
> ```Go
> 	req := NewTask(*queue, *payload)
> 	pretty.Print(req)
> ```

Add a comment about the p~ so that it pops out at everyone!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/7#-:-KhOyEWMH9lk5II79CD1:b-gr8ozl)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
Review status: 0 of 2 files reviewed at latest revision, 5 unresolved discussions.

---

*[appengine/queue.yaml, line 12 at r1](https://reviewable.io:443/reviews/m-lab/etl/7#-KhOy2exwqRVf3Ok53aU:-KhP1bCK37P2YtTRwjuu:b-896fix) ([raw file](https://github.com/m-lab/etl/blob/0365c1e087f528304177ac212d895b6e0d6acf96/appengine/queue.yaml#L12)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

Comment what VM this corresponds to.
</blockquote></details>

Done.

---

*[cmd/task_client/task_client.go, line 23 at r1](https://reviewable.io:443/reviews/m-lab/etl/7#-KhOxWNqfU2xSFaClfkj:-KhP1PUZb5WsC3sipojj:b-uj64sn) ([raw file](https://github.com/m-lab/etl/blob/0365c1e087f528304177ac212d895b6e0d6acf96/cmd/task_client/task_client.go#L23)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

Blank line?
</blockquote></details>

Removed. This ran together with the code below before there was a comment. The blank was for readability, but unnecessary now.

---

*[cmd/task_client/task_client.go, line 24 at r1](https://reviewable.io:443/reviews/m-lab/etl/7#-KhOxYIloHTESTo4GlNB:-KhP1Yale8_QUd6Zbs4-:bwm3udz) ([raw file](https://github.com/m-lab/etl/blob/0365c1e087f528304177ac212d895b6e0d6acf96/cmd/task_client/task_client.go#L24)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

This note isn't super clear.  Ok to leave as is, but if you can make it clearer that would be good.
</blockquote></details>

I've tried to clarify and added a link for more information.

---

*[cmd/task_client/task_client.go, line 44 at r1](https://reviewable.io:443/reviews/m-lab/etl/7#-KhOxFTdBjH-ea9I_-WR:-KhOz7YF7qkQfUIVbCxK:b-896fix) ([raw file](https://github.com/m-lab/etl/blob/0365c1e087f528304177ac212d895b6e0d6acf96/cmd/task_client/task_client.go#L44)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

s/least/lease/
</blockquote></details>

Done.

---

*[cmd/task_client/task_client.go, line 58 at r1](https://reviewable.io:443/reviews/m-lab/etl/7#-KhOxHCxdR9uRsCxY3BT:-KhP1Kej_tMvZMkuWcDP:b-896fix) ([raw file](https://github.com/m-lab/etl/blob/0365c1e087f528304177ac212d895b6e0d6acf96/cmd/task_client/task_client.go#L58)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

Add a comment about the p~ so that it pops out at everyone!
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/7)*
<!-- Sent from Reviewable.io -->
